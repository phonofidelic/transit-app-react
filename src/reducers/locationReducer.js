import { GET_USER_POS, SET_DESTINATION } from '../actiontypes';

const INITIAL_STATE = {
	userPos: null,
	destination: null
};

const locationReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case GET_USER_POS:
			return {
				...state,
				userPos: action.payload,
				uerBbox: {
					ne: {
						lat: action.payload.lat,
						lng: action.payload.lng
					},
					sw: {
						lat: action.payload.lat,
						lng: action.payload.lng
					}
				} 
			};

		case SET_DESTINATION:
			return {
				...state,
				destination: action.payload
			}

		default:
			return state;
	}
}

export default locationReducer;