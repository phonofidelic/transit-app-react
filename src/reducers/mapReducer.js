import { 
	INIT_MAP, 
	MAP_LOADED, 
	UPDATE_MAP_VIEW, 
	SET_DEST_MARKER,
	SET_TRIP_LINE } from '../actiontypes';

const INITIAL_STATE = {
	map: null, 
	mapCenter: null, 
	mapLoaded: false,
	destMarker: null,
	tripLine: null
};

const mapReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {

		case INIT_MAP: 
			return {
				...state,
				map: action.payload
			};

		case MAP_LOADED:
			return {
				...state,
				mapLoaded: true
			}

		case UPDATE_MAP_VIEW:
			return {
				...state,
				mapCenter: action.payload
			};

		case SET_DEST_MARKER:
			return {
				...state,
				destMarker: action.payload
			}

		case SET_TRIP_LINE:
			return {
				...state,
				tripLine: action.payload
			}

		default:
			return state;
	}
}

export default mapReducer;