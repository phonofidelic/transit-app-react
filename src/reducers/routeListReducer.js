import { REQUEST_ROUTES, RECIEVE_ROUTES } from '../actiontypes';

const INITIAL_STATE = { isFetching: false, routes: [] };

const routeListReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case REQUEST_ROUTES:
			return {
				...state,
				isFetching: true
			};

		case RECIEVE_ROUTES:
			return {
				...state,
				isFetching: false,
				routes: action.payload,
				lastUpdated: action.receivedAt
			}

		default:
			return state;
	}
}

export default routeListReducer