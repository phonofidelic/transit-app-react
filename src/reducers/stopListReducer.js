import { FETCH_STOPS } from '../actiontypes';;

export const INITIAL_STATE = {
	stops: []
};

const stopListReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case FETCH_STOPS:
			return {
				...state,
				stops: action.payload
			}

		default:
			return state
	}
}

export default stopListReducer;