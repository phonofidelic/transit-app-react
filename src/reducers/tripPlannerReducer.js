import { 
	SHOW_TRIP_PLANNER,
	HIDE_TRIP_PLANNER,
 	DESTINATION_INPUT_CHANGE ,
 	RECEIVE_AUTOCOMPLETE_RESULTS,
 	SELECT_DESTINATION,
 	RECEIVE_TRIP_DATA
 } from '../actiontypes';

const INITIAL_STATE = {
	showTripPlanner: false,
	destinationInput: null,
	autocompleteResults: [],
	selectedDestination: null,
	tripData: null
};

const tripPlannerReducer = (state = INITIAL_STATE, action) => {
	switch(action.type) {
		case SHOW_TRIP_PLANNER:
			return {
				...state,
				showTripPlanner: true
			}

		case HIDE_TRIP_PLANNER:
			return {
				...state,
				showTripPlanner: false
			}

		case DESTINATION_INPUT_CHANGE:
			return {
				...state,
				destinationInput: action.payload
			}

		case RECEIVE_AUTOCOMPLETE_RESULTS:
			return {
				...state,
				autocompleteResults: action.payload
			}

		case SELECT_DESTINATION:
			return {
				...state,
				selectedDestination: action.payload
			}

		case RECEIVE_TRIP_DATA:
			return {
				...state,
				tripData: action.payload
			}

		default: return state;
	}
}

export default tripPlannerReducer;
