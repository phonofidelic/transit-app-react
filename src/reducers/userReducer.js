import { GET_USER_POS } from '../actiontypes';

const INITIAL_STATE = {userPos: null};

const userReducer = (state = INITIAL_STATE, action) => {
	switch (action.type) {
		case GET_USER_POS:
			return {
				...state,
				userPos: action.payload
			};

		default:
			return state;
	}
}

export default userReducer;