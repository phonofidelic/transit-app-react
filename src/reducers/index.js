import { combineReducers } from 'redux';
import userReducer from './userReducer';
import routeListReducer from './routeListReducer';
import mapReducer from './mapReducer';
import stopListReducer from './stopListReducer';
import tripPlannerReducer from './tripPlannerReducer';

const rootReducer = combineReducers({
	userReducer,
	routeListReducer,
	mapReducer,
	stopListReducer,
	tripPlannerReducer
});

export default rootReducer;