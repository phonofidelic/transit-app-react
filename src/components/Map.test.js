import React from 'react';
import ReactDOM from 'react-dom';
import MapComponent from './Map';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {rootReducer as initialState } from '../reducers';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<MapComponent />', () => {
	it('renders with default props', () => {
		const store = mockStore({initialState});
		const wrapper = shallow(
			<MapComponent store={store} />
		);
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});