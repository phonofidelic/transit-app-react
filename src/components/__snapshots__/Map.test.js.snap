exports[`<MapComponent /> renders with default props 1`] = `
<MapComponent
  fetchNearbyRoutes={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  } />
`;
